mobile-browser-based-version/server/src/logic/federated/async_weights_informant.ts

```diff
@@ -0,0 +1,92 @@
+import { AsyncWeightsBuffer } from './async_weights_buffer'
+
+/**
+ * @remarks
+ * taskID: corresponds to the task that weights correspond to.
+ */
+export class AsyncWeightsInformant {
+    taskID: string;
+    asyncWeightsBuffer: AsyncWeightsBuffer
+
+    // This could rapidly grow so we should have some way to do garbage collection ?
+    // This could be interesting for measures based on gradient history
+    // Maybe we could have some sort of observer on the buffer
+    // so everytime we aggregate, we store the history
+    bufferHistory: Map<number, Map<string, any>>
+
+    round: number = 0
+    currentNumberOfParticipants: number = 0
+    totalNumberOfParticipants: number = 0
+    averageNumberOfParticipants : number = 0
+
+    constructor (taskID: string, asyncWeightsBuffer: AsyncWeightsBuffer) {
+      this.taskID = taskID
+      this.asyncWeightsBuffer = asyncWeightsBuffer
+      this.asyncWeightsBuffer.registerObserver(this)
+      this.printAllInfos()
+    }
+
+    // Update functions
+
+    update () {
+      // DEBUG
+      console.log('Before update')
+      this.printAllInfos()
+
+      this._updateRound()
+      this._updateNumberOfParticipants()
+
+      // DEBUG
+      console.log('After update')
+      this.printAllInfos()
+    }
+
+    _updateRound () {
```

Comment:
In TS you can make private functions by using the `private` tag, just like in java, so no more need to add a _ prefix.